### PR TITLE
Feature/fragment nearby

### PR DESCRIPTION
**Pull request by PassiDel**
Added Nearby shared items, paginated from Firestore with endless scrolling and refreshing.

### CODE REVIEW

File: app/src/main/java/app/wefridge/wefridge/EditFragment.kt [hunk 1]
```diff
@@ -4,37 +4,43 @@ import android.os.Bundle
...
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_edit, container, false)
+        _binding = FragmentEditBinding.inflate(inflater, container, false)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
     }
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/EditFragment.kt:44]: redundant, remove

File: app/src/main/java/app/wefridge/wefridge/EndlessRecyclerOnScrollListener.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                // End has been reached, Do something
+                loading = true
+                onLoadMore.invoke(this)
+            }
+        }
+    }
+
+    fun setLoading(loading: Boolean) {
+        this.loading = loading
+    }
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/EndlessRecyclerOnScrollListener.kt:39]: This necessary?

File: app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    override fun onStart() {
+        super.onStart()
+
+        loadContactInfo()
+    }
+
+    private fun loadContactInfo() {
+        sp = PreferenceManager.getDefaultSharedPreferences(context)
+        val user = FirebaseAuth.getInstance().currentUser!!
+
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt:51]: What happens on fail?

File: app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        binding.bestBy.text = model.bestByDate
+        binding.distance.text = ""
+        binding.additionalInformation.text = model.details
+        binding.owner.text = ""
+
+        val emailIntent = Intent(Intent.ACTION_SENDTO).apply {
+            type = "text/plain"
+            data = Uri.parse("mailto:")
+            // TODO: get mail from model
+            putExtra(Intent.EXTRA_EMAIL, arrayOf("support@arboristapp.com"))
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt:80]: Why arrayOf?

File: app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            putExtra(Intent.EXTRA_SUBJECT, "WeFridge: ${model.content}")
+            putExtra(
+                Intent.EXTRA_TEXT, """Hello,
+
+Shared item: ${model.content}
+
+Best regards,
+$name
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt:88]: Use strings

File: app/src/main/res/navigation/nav_graph.xml [hunk 6]
```diff
@@ -29,9 +29,13 @@
     <fragment
         android:id="@+id/NearbyItemsFragment"
         android:name="app.wefridge.wefridge.NearbyItemFragment"
-        android:label="nearby"
+        android:label="@string/nav_label_nearby"
```
Comment (frcroth) [inline app/src/main/res/navigation/nav_graph.xml:32]: 👍

File: app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt (hunk 3 above)
Comment (PassiDel) [reply app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt:51]: Crash

Comment (frcroth) [reply app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt:51]: :(

File: app/src/main/java/app/wefridge/wefridge/EditFragment.kt (hunk 1 above)
Comment (PassiDel) [reply app/src/main/java/app/wefridge/wefridge/EditFragment.kt:44]:
```suggestion

```

File: app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt (hunk 4 above)
Comment (PassiDel) [reply app/src/main/java/app/wefridge/wefridge/NearbyDetailFragment.kt:80]: [RTFM](https://developer.android.com/guide/components/intents-common#Email)

**Comment by AlexTerhar** [comment]
its fine by me

**Pull request merged**